### ISSUE TITLE
[feature] privacy-check: retry API calls

### ISSUE THREAD

**Issue by ianlewis** [slsa-framework/slsa-github-generator#914]
If a transient error occurs we want to be able to recover without stopping the whole workflow. Maybe retry a couple times if we get back a 500+ error.

**Comment by dibrinsofor** [comment]
like an exponential backoff?

**Comment by ianlewis** [comment]
> @ianlewis like an exponential backoff?

Yeah, something like that. Something that retries a few times. How those retries are spaced out we'll leave to the implementor.
